pkg/cloud/services/s3/s3.go

```diff
@@ -221,11 +218,25 @@ func (s *Service) Delete(m *scope.MachineScope) error {
 
 	s.scope.Info("Deleting S3 object", "bucket", bucket, "key", key)
 
-	_, err = s.S3Client.DeleteObject(&s3.DeleteObjectInput{
+	return s.deleteObject(bucket, key)
+}
+
+func (s *Service) deleteObject(bucket, key string) error {
+	if _, err := s.S3Client.DeleteObject(&s3.DeleteObjectInput{
 		Bucket: aws.String(bucket),
 		Key:    aws.String(key),
-	})
-	if err != nil {
+	}); err != nil {
+		if ptr.Deref(s.scope.Bucket().BestEffortDeleteObjects, false) {
+			if aerr, ok := err.(awserr.Error); ok {
+				switch aerr.Code() {
+				case "Forbidden":
+					fallthrough
+				case "AccessDenied":
```

Comment:
What's the difference between Forbidden and AccessDenied in this case? Why are we ignoring the AccessDenied case?

Reply:
No difference, we are treating them the same (that's why the `fallthrough`). But I'll change this as the linter suggests which should make it more clear.